backend/lib/endpoints/schema/posts.js

```diff
@@ -0,0 +1,66 @@
+const S = require("fluent-schema");
+
+const getPostsSchema = {
+  querystring: S.object()
+    .prop("authorId", S.string())
+    .prop("skip", S.integer())
+    .prop("limit", S.integer()),
+};
+
+const createPostSchema = {
+  body: S.object()
+    .prop("title", S.string().required())
+    .prop("description", S.string().required())
+    .prop("type", S.array().maxItems(10).items(S.string()).required())
+    .prop("shareWith", S.string().required())
+    .prop("needs", S.array().maxItems(10).items(S.string()).required())
+    .prop("comments", S.array().items(S.string()).required())
+    .prop("tags", S.array().items(S.string()).required())
+    .prop("language", S.string().required())
+    .prop("website", S.string())
+    .prop("iosUrl", S.string())
+    .prop("androidUrl", S.string())
+    .prop("media", S.string()),
+};
+
+const getPostByIdSchema = {
+  querystring: S.object()
+    .prop("authorId", S.string())
+    .prop("skip", S.integer())
+    .prop("limit", S.integer()),
+};
+
+const updatePostSchema = {
+  body: S.object()
+    .prop("title", S.string())
+    .prop("description", S.string())
+    .prop("type", S.array().maxItems(10).items(S.string()))
+    .prop("shareWith", S.string())
+    .prop("needs", S.array().maxItems(10).items(S.string()))
+    .prop("comments", S.array().items(S.string()))
+    .prop("tags", S.array().items(S.string()))
+    .prop("language", S.string())
+    .prop("website", S.string())
+    .prop("iosUrl", S.string())
+    .prop("androidUrl", S.string())
+    .prop("media", S.string()),
+  params: S.object().prop("postId", S.string().required()),
+};
+
+const deletePostSchema = {
+  params: S.object().prop("postId", S.string().required()),
+};
+
+const addCommentSchema = {
+  body: S.object().prop("comment", S.string().required()),
+  params: S.object().prop("postId", S.string().required()),
+};
+
+module.exports = {
+  getPostsSchema,
```

Comment:
Minor: perhaps these exports could be ordered in alphabetical order?

Reply:
Great idea, I actually prefer that myself, the alternative is "logical" ordering but usually this gets hard to stay consistent.
I propose to add an ESLint rule for this for all the imports / exports and also for object properties and such, but then I think it's better to create a separate pull request for this

Reply:
@MohamedEl1 just so you know, I'll make a PR for this later today